### PR TITLE
WIP: Added Texture.contain, .cover, .fill

### PR DESCRIPTION
**Pull request by WestLangley**
When a flat texture is set as the scene.background, it stretches to fit by default.

I'm trying to find the right API to support something like the CSS object-fit "cover" and "contain" properties.

(This feature does not have to be supported by adding new methods to `Texture` as I have done. I could just show an example.)

Suggestions welcome...

### CODE REVIEW

**Comment by WestLangley** [comment]
It's alive !

https://raw.githack.com/westlangley/three.js/dev_object_fit/examples/webgl_background_cover.html

File: src/renderers/shaders/ShaderLib/background_frag.glsl.js [hunk 1]
```diff
@@ -5,6 +5,9 @@ varying vec2 vUv;
 
 void main() {
 
+	if ( vUv.x < 0.0 || vUv.x > 1.0 ) discard; // works, but it messes up repeated textures in other use cases
+	if ( vUv.y < 0.0 || vUv.y > 1.0 ) discard;
```
Comment (WestLangley) [inline src/renderers/shaders/ShaderLib/background_frag.glsl.js]: This is required to support "contain". Is it worth supporting contain?

**Comment by donmccurdy** [comment]
Perhaps `Scene` or `SceneUtils` might be a better place for this API? For example:

```js
import { Scene } from 'three';
import { SceneUtils } from 'three/examples/jsm/utils/SceneUtils.js';

const scene = new Scene();
scene.background = texture;
SceneUtils.setBackgroundFit( scene, 'contain', aspect );
```

**Comment by WestLangley** [comment]
@donmccurdy That is probably better...

**Comment by WestLangley** [comment]
@mrdoob Maybe we could have a `SceneBackground` object. You indicated something like that, if I remember correctly. I do not recall the motivation, though. One advantage would be having user-access to the background material.

**Pull request closed without merging**